### DISCUSSION TITLE
react-query-next-experimental's ReactQueryStreamedHydration, no way to fetch on server side components?

### DISCUSSION THREAD

**Discussion by FezVrasta** [TanStack/query#8406] [General]
I was looking at the new `ReactQueryStreamedHydration` examples, but it looks like the idea is to always fetch data from a client side component, never from a server one.

How can I have a server side component fetch some data, render it on the client, and send the cache to the client so that client side queries can make use of it?

One approach I can think of is the following:

```tsx
// app/page.tsx
"use server";

export const Page = () => {
  const queryClient = new QueryClient();
  
  const data = await queryClient.fetchQuery({
    queryKey: ['test'],
    queryFn: async () => (await fetch('https://example.org/api/test')).json()
  });

  return (
    <QueryClientProvider client={queryClient}>
      <ReactQueryStreamedHydration>
        <h1>{data.title}</h1>
        <ClientSideComponent />
      </ReactQueryStreamedHydration>
    </QueryClientProvider>
  );
}
```

```tsx
// components/client-side-component.tsx
"use client"

export const ClientSideComponent = () => {
  // this should not trigger a network request as the same resource was
  // already loaded by the server
  const { data } = useSuspenseQuery({
    queryKey: ['test'],
    queryFn() => (await fetch('https://example.org/api/test')).json(),
  });

  return <p>{data.body}</p>;
}
```

This should allow to share the cache between the client and the server.

I just don't understand why all of this has not been explored or documented anywhere?

**Comment by Vesper0704** [comment]
@TkDodo I've got the same question here. It seems that wrap the component in `ReactQueryStreamedHydration` without prefetching on server doesn't avoid client network request because they are not sharing the query cache ?

I've tested the example [here](https://tanstack.com/query/latest/docs/framework/react/examples/nextjs-suspense-streaming) and the client request is ongoing. What I expected is client and server share the same query cache and no network request is made until the staletime is due.

**Reply by TkDodo** [reply]
when I run the example locally, there is no fetch going out from the client. The data is streamed from the server into the client cache, and data also shows up in the query devtools (which only look at the client cache)
